client/src/App.jsx

```diff
@@ -0,0 +1,77 @@
+import React from 'react';
+import axios from 'axios';
+import SearchBox from './components/SearchBox.jsx';
+import BreakdownBox from './components/BreakdownBox.jsx';
+
+class App extends React.Component {
+  constructor(props) {
+      super(props)
+      this.state = {
+          allReviews: [],
+          avgRating: 0,
+          search: ''
+      }
+      this.getReviews = this.getReviews.bind(this);
+      this.handleClick = this.handleClick.bind(this);
+      this.findAvgReviewRating = this.findAvgReviewRating.bind(this);
+      this.countRecommendedReviews = this.countRecommendedReviews.bind(this);
+      this.handleSearchChange = this.handleSearchChange.bind(this);
+  }
+
+  componentDidMount() {
+    this.getReviews();
+  }
+
+  getReviews() {
+    axios.get('http://localhost:8000/reviews/4894941353195')
+    .then((results) => {
+      this.setState({allReviews: results.data[0].reviews}, () => {
+        this.findAvgReviewRating();
+      });
+    })
+  }
+
+  handleClick() {
+
+  }
+
+  handleSearchChange(e) {
+    e.preventDefault();
+    this.setState({search: e.target.value})
+    console.log(this.state);
+  }
+
+  findAvgReviewRating() {
+    var total = this.state.allReviews.reduce((acc, review) => {return acc + review.rating},0);
+    console.log(this.state);
+    var avg = total/this.state.allReviews.length;
+    if (isNaN(avg)) {
+      avg=0;
```

Comment:
I wonder if it would be better to check before summing the total if there are any reviews. 
Maybe you could have an if statement on this.state.allReviews and if length is equal to zero return ), instead of doing all the calculation and the checking for isNaN

Reply:
great idea, making this change now